### PR TITLE
 treewide: make fetchpatch urls reproducible

### PR DESCRIPTION
**Pull request by pbsds**
## Description of changes

repeat of https://github.com/NixOS/nixpkgs/pull/258431

## Things done

- Built on platform(s)
  - [ ] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [24.11 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2411.section.md) (or backporting [23.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) and [24.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2405.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

### CODE REVIEW

**Comment by pbsds** [comment]
verified with 
```nix
{ pkgs ? import ./. {}, lib ? pkgs.lib, ... }:
let
  patches = with pkgs; [
      libsbml.patches
      webtorrent_desktop.patches
      python311Packages.xgboost.patches
      python313Packages.xgboost.patches
      python311Packages.pymilter.patches
      python313Packages.pymilter.patches
      python311Packages.dazl.patches
      python313Packages.dazl.patches
      mpvScripts.mpv-notify-send.patches
      modern-cpp-kafka.patches
      git-branchless.patches
      [(pkgs.fetchpatch {
        name = "vimus-libmpd-haskell-pull-138.patch";
        url = "https://github.com/vimus/libmpd-haskell/compare/95d3b3bab5858d6d1f0e079d0ab7c2d182336acb...f1cbf247261641565a3937b90721f7955d254c5e.patch";
        sha256 = "Q4fA2J/Tq+WernBo+UIMdj604ILOMlIYkG4Pr046DfM=";
      })]
 ];
in lib.pipe patches [
  lib.flatten
  (builtins.filter lib.isDerivation)
  (builtins.map (drv:
    drv.overrideAttrs (old: {
      name = "invalidated-${old.name}";
    })
  ))
]
```

File: pkgs/development/haskell-modules/configuration-ghc-9.8.x.nix [hunk 1]
```diff
@@ -131,8 +131,10 @@ self: super: {
   hip = appendConfigureFlag "--ghc-options=-fsimpl-tick-factor=200" super.hip;
 
   # Fix build with text-2.x.
-  libmpd = appendPatch (pkgs.fetchpatch
-      { url = "https://github.com/vimus/libmpd-haskell/pull/138.patch";
+  libmpd = appendPatch
```
Comment (sternenseemann) [inline pkgs/development/haskell-modules/configuration-ghc-9.8.x.nix:134]: attribute path of this is `haskell.packages.ghc98.libmpd` for the commit message.

**Comment by sternenseemann** [comment]
I think you should add a `name` argument that matches what `fetchpatch` would use before so that this PR causes no rebuilds (which prevents useless redownloads for users and saves build time).

**Comment by maralorn** [comment]
Thank you, this is a good endeavor. But I like sternis suggestion.

**Comment by pbsds** [comment]
thanks for the bump, i forgot to check back and push after running `nixpkgs-review HEAD` to check for rebuilds

**Comment by pbsds** [comment]
i prefer not to self-merge treewides, anyone willing?

**Pull request merged**